### PR TITLE
python312Packages.shapely_1_8: normalize pname

### PR DESCRIPTION
**Pull request by Stunkymonkey**
## Description of changes

part of #245383

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/shapely/1.8.nix [hunk 1]
```diff
@@ -15,14 +15,15 @@
 }:
 
 buildPythonPackage rec {
-  pname = "Shapely";
+  pname = "shapely";
   version = "1.8.5";
   format = "pyproject";
```
Comment (imincik) [inline pkgs/development/python-modules/shapely/1.8.nix:20]:
```suggestion
  pyproject = true;
```

**Review by imincik — CHANGES_REQUESTED** [review]
I suggested one more improvement, otherwise looks OK. Thanks for this cleanup.

**Comment by imincik** [comment]
Result of `nixpkgs-review pr 297036` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>kicadAddons.kikit</li>
    <li>kicadAddons.kikit-library</li>
    <li>kikit</li>
    <li>kikit.dist</li>
    <li>python311Packages.shapely_1_8</li>
    <li>python311Packages.shapely_1_8.dist</li>
    <li>python312Packages.shapely_1_8</li>
    <li>python312Packages.shapely_1_8.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/shapely/1.8.nix (hunk 1 above)
Comment (imincik) [reply pkgs/development/python-modules/shapely/1.8.nix:20]: @Stunkymonkey , are you OK with adding suggested change to this PR or shell we merge this PR as is ?

**Review by imincik — APPROVED** [review]
Thank you very much. Looks good now. I'll merge in once all tests are finished.

**Comment by imincik** [comment]
Result of `nixpkgs-review pr 297036` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>kicadAddons.kikit</li>
    <li>kicadAddons.kikit-library</li>
    <li>kikit</li>
    <li>kikit.dist</li>
    <li>python311Packages.shapely_1_8</li>
    <li>python311Packages.shapely_1_8.dist</li>
    <li>python312Packages.shapely_1_8</li>
    <li>python312Packages.shapely_1_8.dist</li>
  </ul>
</details>

**Comment by Stunkymonkey** [comment]
@imincik seems finished now. Can we merge?

**Pull request merged**